frontend/src/components/ExercisePage.tsx

```diff
@@ -0,0 +1,90 @@
+import {Exercise} from "../model/Exercise";
+import ExerciseDetails from "./ExerciseDetails";
+import {ChangeEvent, FormEvent, useEffect, useState} from "react";
+import axios from "axios";
+import "../css/ExercisePage.css"
+import {useNavigate} from "react-router-dom";
+
+
+type exercisePageProps = {
+    exercises? : Exercise[]
+    selectedExercisesList: (exercisesList: Exercise[]) => void
+}
+
+export default function ExercisePage(props: exercisePageProps){
+    const [newExercise, setNewExercise] = useState<string>("")
+    const [exercisesList, setExercisesList] = useState<Exercise[]>([])
+
+    const [selectedExercises, setSelectedExercises] = useState<Exercise[]>([])
+
+    useEffect(() => {
+        getExercisesListFromDB()
+    },[])
+
+    const navigate = useNavigate()
+
+    function getExercisesListFromDB(){
+        axios.get("/api/exercises")
+            .then(response => response.data)
+            .then(setExercisesList)
+    }
+
+    function inputNewExercise(event: ChangeEvent<HTMLInputElement>){
+        setNewExercise(event.target.value)
+    }
+
+    function onClickNewExercise(){
+        axios.post("/api/exercises/",
```

Comment:
Wäre es nicht besser den axios Kram in eine Datei auszulagern? Gilt auch für das GET. Schau mal, vlt ein Ticket anlegen, wenn du es nicht bereits hier machen willst

Reply:
Okay, da ich unten sehe das es bereits einen Hook gibt, kann man ja den Inhalt des Hooks und dieser Funktionen in eine `exercise-api-calls.ts` auslagern zB